coalib/output/ConsoleInteraction.py

```diff
@@ -147,16 +147,44 @@ def acquire_actions_and_apply(console_printer,
             action_dict[metadata.name] = action
             metadata_list.append(metadata)
 
+        applied_action = ask_for_action_and_apply(console_printer,
+                                                  section,
+                                                  metadata_list,
+                                                  action_dict,
+                                                  failed_actions,
+                                                  result,
+                                                  file_diff_dict,
+                                                  file_dict)
+
         # User can always choose no action which is guaranteed to succeed
-        if not ask_for_action_and_apply(console_printer,
-                                        section,
-                                        metadata_list,
-                                        action_dict,
-                                        failed_actions,
-                                        result,
-                                        file_diff_dict,
-                                        file_dict):
+        if not applied_action:
             break
+        else:
+            next_actions = applied_action.next_actions
+
+            # FIXME: currently the code doensn't check for
+            # circular dependencies among the next_actions
+            while next_actions:
+                if len(next_actions) == 1:
+                    # Immediately execute the only action
+                    applied_action = directly_apply_action(next_actions[0],
+                                                           console_printer,
+                                                           section,
+                                                           failed_actions,
+                                                           result,
+                                                           file_diff_dict,
+                                                           file_dict)
+                    next_actions = (applied_action.next_actions
+                                    if applied_action else [])
+
+                if len(next_actions) > 1:
+                    result.actions = next_actions
+                    acquire_actions_and_apply(console_printer,
+                                              section,
+                                              file_diff_dict,
+                                              result,
+                                              file_dict)
+                    break
```

Comment:
There is one unintuitive behavior currently, 
Due to the outer `while True:` loop and the inner `while next_actions:` loop, the user is prompted for a particular set of `next_actions` twice (because a "do nothing" action breaks the inner while loop and the outer while loop still persists) which seems unintuitive to me as a chained action behaviour. Not sure if that should be the desired behavior. 

imo the "do nothing" should break out of both the loops or we may have to combine these two while loops somehow.